File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/webapp/src/ds/BpClassificationDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
Ext.define("net.nan21.dnet.module.md.bp.ds.model.BpClassificationDs" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"businessPartner", type:"string"},
		{name:"businessPartnerBO", type:"string"},
		{name:"businessPartnerId", type:"int", useNull:true},
		{name:"classificationCode", type:"string"},
		{name:"classificationId", type:"int", useNull:true},
		{name:"classificationName", type:"string"},
		{name:"classificationSystem", type:"string"},
		{name:"classificationSystemId", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"uuid", type:"string"},
		{name:"version", type:"int", useNull:true}
	]});
Ext.define("net.nan21.dnet.module.md.bp.ds.model.BpClassificationDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"businessPartner", type:"string"},
		{name:"businessPartnerBO", type:"string"},
		{name:"businessPartnerId", type:"int", useNull:true},
		{name:"businessPartnerId_From",type:"int", useNull:true},
		{name:"businessPartnerId_To",type:"int", useNull:true},
		{name:"classificationCode", type:"string"},
		{name:"classificationId", type:"int", useNull:true},
		{name:"classificationId_From",type:"int", useNull:true},
		{name:"classificationId_To",type:"int", useNull:true},
		{name:"classificationName", type:"string"},
		{name:"classificationSystem", type:"string"},
		{name:"classificationSystemId", type:"int", useNull:true},
		{name:"classificationSystemId_From",type:"int", useNull:true},
		{name:"classificationSystemId_To",type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"uuid", type:"string"}
	]
});
Ext.define("net.nan21.dnet.module.md.bp.ds.param.BpClassificationDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
});

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/bp/ds/model/BpClassificationDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.bp.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.Param;
import net.nan21.dnet.core.api.annotation.RefLookup;
import net.nan21.dnet.core.api.annotation.RefLookups;
import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;
import net.nan21.dnet.module.bd.standards.domain.entity.ClassificationItem;
import net.nan21.dnet.module.bd.standards.domain.entity.ClassificationSystem;
import net.nan21.dnet.module.md.bp.domain.entity.BpClassification;
import net.nan21.dnet.module.md.bp.domain.entity.BusinessPartner;

@Ds(entity = BpClassification.class)
@RefLookups({
		@RefLookup(refId = BpClassificationDs.f_businessPartnerId, namedQuery = BusinessPartner.NQ_FIND_BY_CODE, params = {@Param(name = "pCode", field = BpClassificationDs.f_businessPartner)}),
		@RefLookup(refId = BpClassificationDs.f_classificationSystemId, namedQuery = ClassificationSystem.NQ_FIND_BY_CODE, params = {@Param(name = "pCode", field = BpClassificationDs.f_classificationSystem)}),
		@RefLookup(refId = BpClassificationDs.f_classificationId, namedQuery = ClassificationItem.NQ_FIND_BY_SYSCODE_PRIMITIVE, params = {
				@Param(name = "pClassSystemId", field = BpClassificationDs.f_classificationSystemId),
				@Param(name = "pCode", field = BpClassificationDs.f_classificationCode)})})
public class BpClassificationDs extends AbstractAuditableDs<BpClassification> {

	public static final String f_businessPartnerId = "businessPartnerId";
	public static final String f_businessPartner = "businessPartner";
	public static final String f_businessPartnerBO = "businessPartnerBO";
	public static final String f_classificationSystemId = "classificationSystemId";
	public static final String f_classificationSystem = "classificationSystem";
	public static final String f_classificationId = "classificationId";
	public static final String f_classificationCode = "classificationCode";
	public static final String f_classificationName = "classificationName";

	@DsField(join = "left", path = "bp.id")
	private Long businessPartnerId;

	@DsField(join = "left", path = "bp.code")
	private String businessPartner;

	@DsField(join = "left", fetch = false, path = "bp.businessObject")
	private String businessPartnerBO;

	@DsField(join = "left", path = "classSystem.id")
	private Long classificationSystemId;

	@DsField(join = "left", path = "classSystem.code")
	private String classificationSystem;

	@DsField(join = "left", path = "classCode.id")
	private Long classificationId;

	@DsField(join = "left", path = "classCode.code")
	private String classificationCode;

	@DsField(join = "left", path = "classCode.name")
	private String classificationName;

	public BpClassificationDs() {
		super();
	}

	public BpClassificationDs(BpClassification e) {
		super(e);
	}

	public Long getBusinessPartnerId() {
		return this.businessPartnerId;
	}

	public void setBusinessPartnerId(Long businessPartnerId) {
		this.businessPartnerId = businessPartnerId;
	}

	public String getBusinessPartner() {
		return this.businessPartner;
	}

	public void setBusinessPartner(String businessPartner) {
		this.businessPartner = businessPartner;
	}

	public String getBusinessPartnerBO() {
		return this.businessPartnerBO;
	}

	public void setBusinessPartnerBO(String businessPartnerBO) {
		this.businessPartnerBO = businessPartnerBO;
	}

	public Long getClassificationSystemId() {
		return this.classificationSystemId;
	}

	public void setClassificationSystemId(Long classificationSystemId) {
		this.classificationSystemId = classificationSystemId;
	}

	public String getClassificationSystem() {
		return this.classificationSystem;
	}

	public void setClassificationSystem(String classificationSystem) {
		this.classificationSystem = classificationSystem;
	}

	public Long getClassificationId() {
		return this.classificationId;
	}

	public void setClassificationId(Long classificationId) {
		this.classificationId = classificationId;
	}

	public String getClassificationCode() {
		return this.classificationCode;
	}

	public void setClassificationCode(String classificationCode) {
		this.classificationCode = classificationCode;
	}

	public String getClassificationName() {
		return this.classificationName;
	}

	public void setClassificationName(String classificationName) {
		this.classificationName = classificationName;
	}
}
